tasks/list-remaining-migrated-removals.ts

```diff
@@ -0,0 +1,207 @@
+      // hre.log({ jsonData });
+
+      const [signer] = await hre.getSigners();
+      const signerAddress = await signer.getAddress();
+      const { getRemoval } = await import('@/utils/contracts');
+      const removalContract = await getRemoval({
+        hre,
+        signer,
+      });
+      hre.log(`Removal contract address: ${removalContract.address}`);
+      hre.log(`Signer address: ${signerAddress}`);
+      // const fireblocksSigner = removalContract.signer as FireblocksSigner;
+
+      if (!dryRun) {
+        hre.log(
+          chalk.bold.white(
+            `✨ Listing unsold removals for ${jsonData.length} projects...`
+          )
+        );
+      } else {
+        hre.log(
+          chalk.bold.white(
+            `DRY RUN 🌵 Listing unsold removals for ${jsonData.length} projects...`
+          )
+        );
+      }
+
+      const allMigratedRemovalIds = jsonData.flatMap(
+        (project) => project.tokenIds
+      );
+      if (allMigratedRemovalIds.includes(undefined)) {
+        hre.log(
+          chalk.bold.red(
+            `❌ Some migrated projects have undefined token ids. Please check the input file for transaction errors during minting. Exiting...`
+          )
+        );
+        return;
+      }
+
+      hre.log(chalk.white(`👀 Querying unsold removal balances...`));
+
+      const remainingBalanceData = await Promise.all(
+        allMigratedRemovalIds.map(async (tokenId) => {
+          const balance = await removalContract.balanceOf(
+            signerAddress,
+            tokenId
+          );
+          return { tokenId, balance };
+        })
+      );
+      // filter out token ids that have a balance of 0
+      const listableData = remainingBalanceData.filter((data) =>
+        data.balance.gt(0)
+      );
+
+      const listableTokenIds = listableData.map((data) => data.tokenId);
+      const listableBalances = listableData.map((data) => data.balance);
+      // sum the listable balances and convert to ether
+      const totalListableBalance = listableBalances.reduce(
+        (accumulator, balance) => accumulator.add(balance),
+        ethers.BigNumber.from(0)
+      );
+      const totalListableBalanceInEther =
+        ethers.utils.formatEther(totalListableBalance);
+
+      hre.log(
+        chalk.white(
+          `🔎 Found ${listableTokenIds.length} removal tokens with a total listable balance of ${totalListableBalanceInEther} NRTs`
+        )
+      );
+      // if there are no listable token ids, exit
+      if (listableTokenIds.length === 0) {
+        hre.log(
+          chalk.white(
+            `👋 No listable token ids found (no non-zero balances), exiting without listing any removals...`
+          )
+        );
+        return;
+      }
+      if (!dryRun) {
+        hre.log(chalk.white(`🤞 Submitting multicall consign transaction...`));
+        let txResult;
+        let pendingTx: ethers.ContractTransaction;
+        try {
+          pendingTx = await removalContract.multicall(
```

Comment:
```suggestion
          const functionOrDryRun = dryRun ? removalContract.multicall : removalContract.callStatic.multicall;
          pendingTx = await functionOrDryRun(
```

Then delete the if/else blocks. Also wonder if we should always call callStatic before actually executing regardless if the flag is used.

Reply:
the issue with deleting the if else blocks is that the return value from the dry run is void, so we don't actually get a pending transaction and would have to write a bunch of more granular logic to avoid performing the rest of the transaction awaiting, file writing, etc. that is different between the two scenarios